interface/backend/submission.py

```diff
@@ -11,10 +16,24 @@
 def handle_submission(file, assignment, user):
     log.debug(f'Submission {file.name} received')
 
-    submission = assignment.submission_set.create(
-        user=user,
-        archive_size=file.size,
-    )
+    with transaction.atomic():
+        Assignment.objects.select_for_update().get(pk=assignment.pk)
```

Comment:
Please add a comment here, something like "Computing time from the last upload by this user. We lock the assignment row to prevent simultaneous uploads (race condition)."